src/main/scala/xiangshan/cache/dcache/data/BankedDataArray.scala

```diff
@@ -160,14 +157,15 @@ class DataSRAMBank(index: Int)(implicit p: Parameters) extends DCacheModule {
 
   val r_way_en_reg = RegNext(io.r.way_en)
 
-  val w_reg = RegNext(io.w)
+  // external controls do not read and write at the same time
+  val w_info = io.w
```

Comment:
Is it safe to reduce one cycle?

Reply:
It does not reduce one cycle, because the coming signal is the result of `RegNext` outside this module. 

On the one hand, this modification unifies the cycles inside the module, that is, the data is written when the request comes. The operation of `RegNext` is exposed outside the module. On the other hand, the signal after `RegNext` is also needed outside the module.